extensions/mssql/src/objectExplorerNodeProvider/webhdfs.ts

```diff
@@ -575,21 +645,51 @@ export class WebHDFS {
 		this.checkArgDefined('path', path);
 
 		let endpoint = this.getOperationEndpoint('open', path, opts);
-		let params = Object.assign(
+		let params: request.OptionsWithUrl = Object.assign(
 			{
 				method: 'GET',
 				url: endpoint,
 				json: true
 			},
 			this._requestParams
 		);
+		if (!this._requestParams.isKerberos) {
+			return <fs.ReadStream><any>this.doCreateReadStream(params);
```

Comment:
what does <fs.ReadStream><any> do here?

Reply:
It just lets upstream callers treat this as a ReadStream so they know what events to expect via the pipe